### PR TITLE
cleanup(libsinsp): adjust setting max_thread_table_size

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

I believe we just need this small change to expose setting max_thread_table_size via configs in Falco. Will create the Falco PR once we are past the next release and have bumped libs.

**Which issue(s) this PR fixes**:

https://github.com/falcosecurity/libs/issues/1621 @jasondellaluce and @FedeDP 

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by FedeDP — APPROVED** [review]
/approve

File: userspace/libsinsp/threadinfo.cpp [hunk 1]
```diff
@@ -2178,7 +2178,7 @@ threadinfo_map_t::ptr_t sinsp_thread_manager::find_thread(int64_t tid, bool look
 
 void sinsp_thread_manager::set_max_thread_table_size(uint32_t value)
 {
-    m_max_thread_table_size = std::min(value, m_thread_table_absolute_max_size);
+    m_max_thread_table_size = value;
```
Comment (FedeDP) [inline userspace/libsinsp/threadinfo.cpp:2181]: Am not sure whether we want to retain some sort of `max` value here.

Comment (incertum) [reply userspace/libsinsp/threadinfo.cpp:2181]: How would we choose the absolute new max value in a way that wouldn't limit flexibility?

Comment (FedeDP) [reply userspace/libsinsp/threadinfo.cpp:2181]: Yes fair point! I am not sure , like 10x default value? But of course that would still be an hard limit...

Comment (incertum) [reply userspace/libsinsp/threadinfo.cpp:2181]: We have UINT32 max value as super hard limit ...

Comment (FedeDP) [reply userspace/libsinsp/threadinfo.cpp:2181]: Let's go as is then! I was just thinking loud :)

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**